File: cases/taint/javascript/deep-relay-chain-positive/DeepRelayChainPositive.js
function dfb_source() { // DFB-SOURCE: deep-relay-chain-input
  return "tainted";
}

function dfb_sink(value) {} // DFB-SINK: deep-relay-chain-sink

function relay6(value) { // DFB-WITNESS: deep-relay-chain-hop-six
  dfb_sink(value);
}

function relay5(value) {
  relay6(value);
}

function relay4(value) {
  relay5(value);
}

function relay3(value) {
  relay4(value);
}

function relay2(value) {
  relay3(value);
}

function relay1(value) { // DFB-WITNESS: deep-relay-chain-hop-one
  relay2(value);
}

function run() {
  relay1(dfb_source());
}

File: cases/taint/csharp/deep-relay-chain-positive/DeepRelayChainPositive.cs
namespace DataFlowBench;

static class DeepRelayChainPositive
{
    static string dfb_source() // DFB-SOURCE: deep-relay-chain-input
    {
        return "tainted";
    }

    static void dfb_sink(string value) { } // DFB-SINK: deep-relay-chain-sink

    static void Relay6(string value) // DFB-WITNESS: deep-relay-chain-hop-six
    {
        dfb_sink(value);
    }

    static void Relay5(string value)
    {
        Relay6(value);
    }

    static void Relay4(string value)
    {
        Relay5(value);
    }

    static void Relay3(string value)
    {
        Relay4(value);
    }

    static void Relay2(string value)
    {
        Relay3(value);
    }

    static void Relay1(string value) // DFB-WITNESS: deep-relay-chain-hop-one
    {
        Relay2(value);
    }

    static void Run()
    {
        Relay1(dfb_source());
    }
}
